stdlib/sys/process.gr

```diff
@@ -61,19 +78,130 @@ enum Signal {
 
 // Access command line arguments
 // @returns Array<String> The positional string arguments to the process
-import foreign wasm argv : () -> Array<String> from "stdlib-external/sys"
+export let argv = () => {
+  let argcPtr = Memory.malloc(8n)
+  let argvBufSizePtr = argcPtr + 4n
+
+  let mut err = Wasi.args_sizes_get(argcPtr, argvBufSizePtr)
+  if (err != Wasi._ESUCCESS) {
+    Memory.free(argcPtr)
+    throwError(Errors._GRAIN_ERR_SYSTEM, WasmI32.fromGrain(tagSimpleNumber(err)), 0n)
+  }
+
+  let argc = WasmI32.load(argcPtr, 0n)
+  let argvBufSize = WasmI32.load(argvBufSizePtr, 0n)
+
+  let argvPtr = Memory.malloc(argc * 4n)
+  let argvBufPtr = Memory.malloc(argvBufSize)
+
+  err = Wasi.args_get(argvPtr, argvBufPtr)
+  if (err != Wasi._ESUCCESS) {
+    Memory.free(argcPtr)
+    Memory.free(argvPtr)
+    Memory.free(argvBufPtr)
+    throwError(Errors._GRAIN_ERR_SYSTEM, WasmI32.fromGrain(tagSimpleNumber(err)), 0n)
+  }
+
+  let arr = allocateArray(argc)
+
+  let argsLength = argc * 4n
+  for (let mut i = 0n; i < argsLength; i += 4n) {
+    let strPtr = WasmI32.load(argvPtr + i, 0n)
+    let mut strLength = 0n
+    while (WasmI32.load8U(strPtr + strLength, 0n) != 0n) {
+      strLength += 1n
+    }
+
+    let grainStrPtr = allocateString(strLength)
+    Memory.copy(grainStrPtr + 8n, strPtr, strLength)
+
+    WasmI32.store(arr + i, grainStrPtr, 8n)
+  }
+
+  Memory.free(argcPtr)
+  Memory.free(argvPtr)
+  Memory.free(argvBufPtr)
+
+  WasmI32.toGrain(arr): Array<String>
```

Comment:
Do we still want this to be an array instead of a List now that it is in Grain?

Reply:
I'm thinking no, specifically because the arguments are positional? OCaml seems to also keep them as an array: https://caml.inria.fr/pub/docs/manual-ocaml/libref/Sys.html